pkg/virt-handler/vm.go

```diff
@@ -2358,24 +2359,32 @@ func (d *VirtualMachineController) getLauncherClientInfo(vmi *v1.VirtualMachineI
 	return launcherInfo
 }
 
-func (d *VirtualMachineController) vmUpdateHelperMigrationSource(origVMI *v1.VirtualMachineInstance) error {
-	client, err := d.getLauncherClient(origVMI)
-	if err != nil {
-		return fmt.Errorf(unableCreateVirtLauncherConnectionFmt, err)
-	}
+func isMigrationInProgress(vmi *v1.VirtualMachineInstance, domain *api.Domain) bool {
+	var domainMigrationMetadata *api.MigrationMetadata
 
-	vmi := origVMI.DeepCopy()
+	if domain == nil ||
+		vmi.Status.MigrationState == nil ||
+		domain.Spec.Metadata.KubeVirt.Migration == nil {
+		return false
+	}
+	domainMigrationMetadata = domain.Spec.Metadata.KubeVirt.Migration
 
-	err = hostdisk.ReplacePVCByHostDisk(vmi)
-	if err != nil {
-		return err
+	if vmi.Status.MigrationState.MigrationUID == domainMigrationMetadata.UID &&
+		domainMigrationMetadata.StartTimestamp != nil {
+		return true
 	}
+	return false
+}
 
-	err = d.handleSourceMigrationProxy(vmi)
+func (d *VirtualMachineController) vmUpdateHelperMigrationSource(origVMI *v1.VirtualMachineInstance, domain *api.Domain) error {
+
+	client, err := d.getLauncherClient(origVMI)
 	if err != nil {
-		return fmt.Errorf("failed to handle migration proxy: %v", err)
+		return fmt.Errorf(unableCreateVirtLauncherConnectionFmt, err)
 	}
 
+	vmi := origVMI.DeepCopy()
```

Comment:
Nit: This is probably only relevant for `ReplacePVCByHostDisk`

Reply:
Done